Fix: Display Care Loading Icon Instead of "No Assets Found" on Asset Tab Revisit on advanced filter

Mahendar0701 opened the pull request:
## Proposed Changes

- Fixes #9202 

- Replaced the "No assets found" message with the Care loading icon when advanced filters are applied, ensuring better user experience.

![image](https://github.com/user-attachments/assets/8e9cb06c-b352-4f62-8fbe-2ac51e25b6a7)

- Added margin for sample test history

![image](https://github.com/user-attachments/assets/3457e67d-d2bf-4b96-ad65-0e1483be2e9c)


@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [ ] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced state management for asset data in the AssetsList component.
	- Improved loading and rendering logic to provide clearer user feedback during asset loading.

- **Bug Fixes**
	- Streamlined asset existence checks, ensuring accurate display of asset information or "No Assets Found" messages.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Mahendar0701 committed:
corrected loading message

Jacobjeevan commented on src/components/Assets/AssetsList.tsx:
```diff
@@ -210,7 +210,7 @@ const AssetsList = () => {
 
   let manageAssets = null;
-  if (loading) {
+  if (loading || !assetsExist) {
```
Else condition would never be triggered; i.e. would result in loading being displayed when there are no results after filter is applied.

Mahendar0701 committed:
updated loading logic

Mahendar0701 committed:
Merge branch 'develop' into relevant-loading-icon-in-assetspage

Mahendar0701 commented:
Hey @Jacobjeevan  i have done the required changes . Once look into it

Mahendar0701 committed:
Merge branch 'develop' into relevant-loading-icon-in-assetspage

rithviknishad reviewed (changes requested):
This issue doesn't seem to be happening for tabs other than Assets tab. Most of the other tabs are having the similar implementation. I'd recommend actually debugging why that "no assets found" is coming in the first place as how it's mentioned in the issue. 

Using timer to solve this is more of a hack that hides the root cause of the issue and is not the right solution.

Ideally the `loading` state provided by the `useQuery` should be enough to show the Loading state and if not loading, `data.results.length` being 0 would convey if it's actually empty assets for the applied filters.

Mahendar0701 committed:
updated loading state without using delay

Mahendar0701 committed:
Merge branch 'develop' into relevant-loading-icon-in-assetspage

Mahendar0701 commented:
> This issue doesn't seem to be happening for tabs other than Assets tab. Most of the other tabs are having the similar implementation. I'd recommend actually debugging why that "no assets found" is coming in the first place as how it's mentioned in the issue.

Yes, The issue is not occurring in other tabs; it is only happening on the Assets page. I have checked the code for the other tabs, and they have a similar implementation. However, in `AssetsList.tsx`, we are not using destructuring to assign `data `in `useQuery`. In contrast, in the other pages, `data ` is being destructured, and the loading logic is also implemented using `data?.count`.

So, I have updated the `AssetsList.tsx` loading logic to match the implementation of the other tabs.
> 
> Using timer to solve this is more of a hack that hides the root cause of the issue and is not the right solution.
> 
I have removed it
> Ideally the `loading` state provided by the `useQuery` should be enough to show the Loading state and if not loading, `data.results.length` being 0 would convey if it's actually empty assets for the applied filters.

On page revisit, the data is initially undefined, which is why the issue is occurring. This condition was not properly handled 
 along with the loading state in the if statement in `AssetsList.tsx`, like it was done in other tabs.

Mahendar0701 commented on src/components/Assets/AssetsList.tsx:
```diff
@@ -210,13 +213,13 @@ const AssetsList = () => {
     );
 
   let manageAssets = null;
-  if (loading) {
+  if (loading || !data) {
     manageAssets = (
       <div className="col-span-3 w-full py-8 text-center">
         <Loading />
       </div>
     );
```
The `data`  is `undefined` initially on page revisit with advanced filter. Added to if condition to resolve issue, and also it is implemented like this in other tabs.

rithviknishad commented on src/components/Patient/SampleDetails.tsx:
```diff
@@ -264,7 +264,10 @@ export const SampleDetails = ({ id }: DetailRoute) => {
   const renderFlow = (flow: FlowModel) => {
     return (
-      <Card key={flow.id}>
+      <Card
+        key={flow.id}
+        className="mb-2 shadow-sm border border-secondary-300"
+      >
```
Revert changes unrelated to this PR

Mahendar0701 replied:
Okay

rithviknishad commented on src/components/Assets/AssetsList.tsx:
```diff
@@ -210,13 +213,13 @@ const AssetsList = () => {
       </div>
     );
-  } else if (assetsExist) {
+  } else if (data?.count) {
```
the previous if block already checks if `data` is undefined or falsey.

```suggestion
  } else if (data.count) {
```

rithviknishad reviewed (commented):
LGTM; just a minor correction

Mahendar0701 committed:
changes added

Mahendar0701 committed:
Merge branch 'develop' into relevant-loading-icon-in-assetspage

rithviknishad commented:
@Jacobjeevan review required

rithviknishad commented:
Closing as root issue was fixed in #9222
